src/map.cpp

```diff
@@ -137,4 +138,93 @@ Map::~Map()
     close(mapfd_);
 }
 
+void MapManager::Add(std::unique_ptr<IMap> map)
+{
+  auto name = map->name_;
+  if (name.empty())
+    throw std::runtime_error("Map without name cannot be stored");
+
+  auto count = maps_by_name_.count(name);
+  if (count > 0)
+    throw std::runtime_error("Map with name: @" + name + " already exists");
+
+  std::shared_ptr<IMap> smap = std::move(map);
```

Comment:
A little strange to use shared_ptrs here. Might be semantically more correct to store 1 unique ptr and the rest raw pointers, as MapManager already "owns" the map.